{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "55028b0a",
   "metadata": {},
   "source": [
    "# tss detection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51218830",
   "metadata": {},
   "outputs": [],
   "source": [
    "# #!/bin/bash\n",
    "# #Input flags for batch files, do not run on notebook\n",
    "# while getopts 'i:' flag; do\n",
    "#   case \"${flag}\" in\n",
    "#       i) BAMFiles=\"${OPTARG}\";;\n",
    "#       ?) echo \"I don't know what flag is this\" ;;\n",
    "#            esac\n",
    "#        done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7219e8d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#temporary assignment of input variables for notebook, remove or comment out in bash file\n",
    "BAMFiles=input_bam_files/BAM_filesLUZ100"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc886825",
   "metadata": {},
   "source": [
    "# function to determine peaks using termseq_peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "67c07196",
   "metadata": {},
   "outputs": [],
   "source": [
    "# function to determine peaks using termseq_peaks\n",
    "determinePeaks(){\n",
    "    directoryBAMFiles=$1\n",
    "    outputDirectory=$2\n",
    "    ends=$3\n",
    "    significanceLevel=$(4:-0.001)\n",
    "    \n",
    "    for file in $directoryBAMFiles/*.bam\n",
    "    do\n",
    "        # Extract file name of BAM file to use for other files\n",
    "        filename_extended=${file##*/}\n",
    "        searchstring=\".sorted.bam\"\n",
    "        filename=${filename_extended%$searchstring*}\n",
    "\n",
    "        # Create directory based on file name\n",
    "        mkdir ${outputDirectory}\n",
    "        mkdir ${outputDirectory}/${filename}\n",
    "\n",
    "        if [[ ends==\"5\" ]]\n",
    "        then\n",
    "            # Get genome coverage of bam files for both + and - strands. Used for termseq_peaks\n",
    "            bedtools genomecov \\\n",
    "            -ibam ${file} \\\n",
    "            -bga \\\n",
    "            -5 \\\n",
    "            -strand + > ${outputDirectory}/${filename}/${filename}.plus.bedgraph\n",
    "\n",
    "            bedtools genomecov \\\n",
    "            -ibam ${file} \\\n",
    "            -bga \\\n",
    "            -5 \\\n",
    "            -strand - > ${outputDirectory}/${filename}/${filename}.minus.bedgraph\n",
    "            elif [[ ends==\"3\" ]]\n",
    "        then\n",
    "            # Get genome coverage of bam files for both + and - strands. Used for termseq_peaks\n",
    "            bedtools genomecov \\\n",
    "            -ibam ${file} \\\n",
    "            -bga \\\n",
    "            -3 \\\n",
    "            -strand + > ${outputDirectory}/${filename}/${filename}.plus.bedgraph\n",
    "\n",
    "            bedtools genomecov \\\n",
    "            -ibam ${file} \\\n",
    "            -bga \\\n",
    "            -3 \\\n",
    "            -strand - > ${outputDirectory}/${filename}/${filename}.minus.bedgraph\n",
    "        else\n",
    "            echo \"error: wrong ends flag\"\n",
    "        fi\n",
    "        \n",
    "\n",
    "        # Use termseq_peaks to get peaks  \n",
    "        termseq_peaks ${outputDirectory}/${filename}/${filename}.plus.bedgraph \\\n",
    "        ${outputDirectory}/${filename}/${filename}.plus.bedgraph --peaks ${outputDirectory}/${filename}/${filename}.plus.peaks \\\n",
    "        --strand + -t $significanceLevel\n",
    "\n",
    "        termseq_peaks ${outputDirectory}/${filename}/${filename}.minus.bedgraph \\\n",
    "        ${outputDirectory}/${filename}/${filename}.minus.bedgraph --peaks ${outputDirectory}/${filename}/${filename}.minus.peaks \\\n",
    "        --strand - -t $significanceLevel\n",
    "\n",
    "        # Add coverage information\n",
    "        bedtools intersect \\\n",
    "          -wao \\\n",
    "          -a ${outputDirectory}/${filename}/${filename}.plus.peaks.oracle.narrowPeak \\\n",
    "          -b ${outputDirectory}/${filename}/${filename}.plus.bedgraph \\\n",
    "          > ${outputDirectory}/${filename}/${filename}.plus.peaks.oracle.narrowPeak.counts\n",
    "\n",
    "        bedtools intersect \\\n",
    "          -wao \\\n",
    "          -a ${outputDirectory}/${filename}/${filename}.minus.peaks.oracle.narrowPeak \\\n",
    "          -b ${outputDirectory}/${filename}/${filename}.minus.bedgraph \\\n",
    "          > ${outputDirectory}/${filename}/${filename}.minus.peaks.oracle.narrowPeak.counts\n",
    "\n",
    "          # Add total mapped reads\n",
    "          total_mapped=$(samtools view -c -F4 $directoryBAMFiles/$filename_extended)\n",
    "\n",
    "        awk '{print $1, $2, $3, $4, $5, $6, $7, $8 ,$9, $10, $11, $12, $13, $14, $15, '$total_mapped'}' \\\n",
    "        ${outputDirectory}/${filename}/${filename}.plus.peaks.oracle.narrowPeak.counts \\\n",
    "        > ${outputDirectory}/${filename}/${filename}.plus.peaks.oracle.narrowPeak.counts.normalized\n",
    "\n",
    "\n",
    "        awk '{print $1, $2, $3, $4, $5, $6, $7, $8 ,$9, $10, $11, $12, $13, $14, $15, '$total_mapped'}' \\\n",
    "        ${outputDirectory}/${filename}/${filename}.minus.peaks.oracle.narrowPeak.counts \\\n",
    "        > ${outputDirectory}/${filename}/${filename}.minus.peaks.oracle.narrowPeak.counts.normalized\n",
    "done\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c874cebe",
   "metadata": {},
   "source": [
    "# TSS detection and clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "872488a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "mkdir: cannot create directory ‘tssLUZ100/infectedControl-LUZ100_full-genome’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.dwnst74i.tmp\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/infectedControl-LUZ100_full-genome/infectedControl-LUZ100_full-genome.plus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/infectedControl-LUZ100_full-genome/infectedControl-LUZ100_full-genome.plus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.i2pwk6d7.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.mss98ect.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.i2pwk6d7.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.mss98ect.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [1.89 0.20 0.91 0.62]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.sg4hpirc.tmp\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/infectedControl-LUZ100_full-genome/infectedControl-LUZ100_full-genome.minus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/infectedControl-LUZ100_full-genome/infectedControl-LUZ100_full-genome.minus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.h0xe1_2c.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.x2gt4doy.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.h0xe1_2c.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.x2gt4doy.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [1.36 0.20 0.94 0.28]\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "mkdir: cannot create directory ‘tssLUZ100/infectedControl-NEB_control_spike_in’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.f8xfpfeh.tmp\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:tssLUZ100/infectedControl-NEB_control_spike_in/infectedControl-NEB_control_spike_in.plus.bedgraph\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:tssLUZ100/infectedControl-NEB_control_spike_in/infectedControl-NEB_control_spike_in.plus.bedgraph\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.vb8qhafh.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.unosj2zr.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.vb8qhafh.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.unosj2zr.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [0.62 0.53 0.99 0.34]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.59ti7own.tmp\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yostrich/anaconda3/bin/termseq_peaks\", line 63, in <module>\n",
      "    idr_peak_calls(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 527, in idr_peak_calls\n",
      "    oracle = mg.call_peaks().peaks_to_bed().moveto(oracle_fn)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 133, in peaks_to_bed\n",
      "    x = merge_narrowbed(x, self.strand)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 589, in merge_narrowbed\n",
      "    x = (peaks.cut([0, 1, 2, 4]).merge(c=4, **additional_kwargs)).to_dataframe()\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 923, in decorated\n",
      "    result = method(self, *args, **kwargs)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 402, in wrapped\n",
      "    stream = call_bedtools(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/helpers.py\", line 460, in call_bedtools\n",
      "    raise BEDToolsError(subprocess.list2cmdline(cmds), stderr)\n",
      "pybedtools.helpers.BEDToolsError: \n",
      "Command was:\n",
      "\n",
      "\tbedtools merge -o max -i /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.09wolk4q.tmp -d 1 -c 4\n",
      "\n",
      "Error message was:\n",
      "\n",
      "*****\n",
      "***** ERROR: Requested column 4, but database file /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.09wolk4q.tmp only has fields 1 - 0.\n",
      "\n",
      "Error: Unable to open file tssLUZ100/infectedControl-NEB_control_spike_in/infectedControl-NEB_control_spike_in.minus.peaks.oracle.narrowPeak. Exiting.\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "mkdir: cannot create directory ‘tssLUZ100/infectedControl-PaLo41_Reference’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.gmc6lynu.tmp\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/infectedControl-PaLo41_Reference/infectedControl-PaLo41_Reference.plus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/infectedControl-PaLo41_Reference/infectedControl-PaLo41_Reference.plus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.04e4x1z_.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.2zv9r2hk.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.04e4x1z_.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.2zv9r2hk.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.39 0.20 0.87 0.16]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.6hcmbkr2.tmp\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/infectedControl-PaLo41_Reference/infectedControl-PaLo41_Reference.minus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/infectedControl-PaLo41_Reference/infectedControl-PaLo41_Reference.minus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.05k7rxph.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.1un2rs2i.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.05k7rxph.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.1un2rs2i.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.38 0.20 0.87 0.16]\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "mkdir: cannot create directory ‘tssLUZ100/infectedEnriched-LUZ100_full-genome’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.axlenfip.tmp\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/infectedEnriched-LUZ100_full-genome/infectedEnriched-LUZ100_full-genome.plus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/infectedEnriched-LUZ100_full-genome/infectedEnriched-LUZ100_full-genome.plus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.yij1ywqs.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.uowcvjbk.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.yij1ywqs.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.uowcvjbk.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.05 0.20 0.97 0.43]\n",
      "INFO:root:Unioning bedgraphs...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.i881maau.tmp\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/infectedEnriched-LUZ100_full-genome/infectedEnriched-LUZ100_full-genome.minus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/infectedEnriched-LUZ100_full-genome/infectedEnriched-LUZ100_full-genome.minus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.k_rv7lvx.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.vw5xfqog.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.k_rv7lvx.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.vw5xfqog.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [1.22 0.20 0.94 0.58]\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.zw6anp1g.tmp\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:tssLUZ100/infectedEnriched-NEB_control_spike_in/infectedEnriched-NEB_control_spike_in.plus.bedgraph\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:tssLUZ100/infectedEnriched-NEB_control_spike_in/infectedEnriched-NEB_control_spike_in.plus.bedgraph\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.oe__g9d2.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.02f9ak9r.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.oe__g9d2.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.02f9ak9r.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [0.46 0.29 0.99 0.62]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.plv61lej.tmp\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yostrich/anaconda3/bin/termseq_peaks\", line 63, in <module>\n",
      "    idr_peak_calls(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 527, in idr_peak_calls\n",
      "    oracle = mg.call_peaks().peaks_to_bed().moveto(oracle_fn)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 133, in peaks_to_bed\n",
      "    x = merge_narrowbed(x, self.strand)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 589, in merge_narrowbed\n",
      "    x = (peaks.cut([0, 1, 2, 4]).merge(c=4, **additional_kwargs)).to_dataframe()\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 923, in decorated\n",
      "    result = method(self, *args, **kwargs)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 402, in wrapped\n",
      "    stream = call_bedtools(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/helpers.py\", line 460, in call_bedtools\n",
      "    raise BEDToolsError(subprocess.list2cmdline(cmds), stderr)\n",
      "pybedtools.helpers.BEDToolsError: \n",
      "Command was:\n",
      "\n",
      "\tbedtools merge -o max -i /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.v43qx2la.tmp -d 1 -c 4\n",
      "\n",
      "Error message was:\n",
      "\n",
      "*****\n",
      "***** ERROR: Requested column 4, but database file /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.v43qx2la.tmp only has fields 1 - 0.\n",
      "\n",
      "Error: Unable to open file tssLUZ100/infectedEnriched-NEB_control_spike_in/infectedEnriched-NEB_control_spike_in.minus.peaks.oracle.narrowPeak. Exiting.\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.jswx_iig.tmp\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/infectedEnriched-PaLo41_Reference/infectedEnriched-PaLo41_Reference.plus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/infectedEnriched-PaLo41_Reference/infectedEnriched-PaLo41_Reference.plus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.nz99h9f3.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.d2x77atr.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.nz99h9f3.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.d2x77atr.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [1.08 0.34 0.99 0.63]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.bvejk0ez.tmp\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/infectedEnriched-PaLo41_Reference/infectedEnriched-PaLo41_Reference.minus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/infectedEnriched-PaLo41_Reference/infectedEnriched-PaLo41_Reference.minus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.bssggel5.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.4rp62hpu.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.bssggel5.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.4rp62hpu.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.13 0.20 0.99 0.36]\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.4y86iz9d.tmp\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:tssLUZ100/uninfectedControl--NEB_control_spike_in/uninfectedControl--NEB_control_spike_in.plus.bedgraph\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:tssLUZ100/uninfectedControl--NEB_control_spike_in/uninfectedControl--NEB_control_spike_in.plus.bedgraph\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.mmtarsv0.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.tghza0l3.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.mmtarsv0.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.tghza0l3.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.05 0.20 0.88 0.11]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.sl204i_t.tmp\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yostrich/anaconda3/bin/termseq_peaks\", line 63, in <module>\n",
      "    idr_peak_calls(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 527, in idr_peak_calls\n",
      "    oracle = mg.call_peaks().peaks_to_bed().moveto(oracle_fn)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 133, in peaks_to_bed\n",
      "    x = merge_narrowbed(x, self.strand)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 589, in merge_narrowbed\n",
      "    x = (peaks.cut([0, 1, 2, 4]).merge(c=4, **additional_kwargs)).to_dataframe()\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 923, in decorated\n",
      "    result = method(self, *args, **kwargs)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 402, in wrapped\n",
      "    stream = call_bedtools(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/helpers.py\", line 460, in call_bedtools\n",
      "    raise BEDToolsError(subprocess.list2cmdline(cmds), stderr)\n",
      "pybedtools.helpers.BEDToolsError: \n",
      "Command was:\n",
      "\n",
      "\tbedtools merge -o max -i /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.a5ah131e.tmp -d 1 -c 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Error message was:\n",
      "\n",
      "*****\n",
      "***** ERROR: Requested column 4, but database file /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.a5ah131e.tmp only has fields 1 - 0.\n",
      "\n",
      "Error: Unable to open file tssLUZ100/uninfectedControl--NEB_control_spike_in/uninfectedControl--NEB_control_spike_in.minus.peaks.oracle.narrowPeak. Exiting.\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.vxr7lbzj.tmp\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/uninfectedControl-LUZ100_full-genome/uninfectedControl-LUZ100_full-genome.plus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/uninfectedControl-LUZ100_full-genome/uninfectedControl-LUZ100_full-genome.plus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.cn77ogd8.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.ty2l4lwm.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.cn77ogd8.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.ty2l4lwm.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [0.10 0.20 0.10 0.99]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.fvph6ws3.tmp\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yostrich/anaconda3/bin/termseq_peaks\", line 63, in <module>\n",
      "    idr_peak_calls(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 527, in idr_peak_calls\n",
      "    oracle = mg.call_peaks().peaks_to_bed().moveto(oracle_fn)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 133, in peaks_to_bed\n",
      "    x = merge_narrowbed(x, self.strand)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 589, in merge_narrowbed\n",
      "    x = (peaks.cut([0, 1, 2, 4]).merge(c=4, **additional_kwargs)).to_dataframe()\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 923, in decorated\n",
      "    result = method(self, *args, **kwargs)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 402, in wrapped\n",
      "    stream = call_bedtools(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/helpers.py\", line 460, in call_bedtools\n",
      "    raise BEDToolsError(subprocess.list2cmdline(cmds), stderr)\n",
      "pybedtools.helpers.BEDToolsError: \n",
      "Command was:\n",
      "\n",
      "\tbedtools merge -o max -i /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.8vl9u6cz.tmp -d 1 -c 4\n",
      "\n",
      "Error message was:\n",
      "\n",
      "*****\n",
      "***** ERROR: Requested column 4, but database file /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.8vl9u6cz.tmp only has fields 1 - 0.\n",
      "\n",
      "Error: Unable to open file tssLUZ100/uninfectedControl-LUZ100_full-genome/uninfectedControl-LUZ100_full-genome.minus.peaks.oracle.narrowPeak. Exiting.\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.p5c2mklw.tmp\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/uninfectedControl-PaLo41_Reference/uninfectedControl-PaLo41_Reference.plus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/uninfectedControl-PaLo41_Reference/uninfectedControl-PaLo41_Reference.plus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.ams44clw.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.08p54in3.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.ams44clw.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.08p54in3.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.38 0.20 0.89 0.14]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.dl1gbmqr.tmp\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/uninfectedControl-PaLo41_Reference/uninfectedControl-PaLo41_Reference.minus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/uninfectedControl-PaLo41_Reference/uninfectedControl-PaLo41_Reference.minus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.79ri8_gg.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.eb4d_yaf.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.79ri8_gg.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.eb4d_yaf.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.43 0.20 0.85 0.15]\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.nv6ezq25.tmp\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/uninfectedEnriched-LUZ100_full-genome/uninfectedEnriched-LUZ100_full-genome.plus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:tssLUZ100/uninfectedEnriched-LUZ100_full-genome/uninfectedEnriched-LUZ100_full-genome.plus.bedgraph\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.whi_yacb.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.oiqpl75i.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.whi_yacb.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.oiqpl75i.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [0.13 0.20 0.50 0.99]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.1jlp0vur.tmp\n",
      "INFO:root:Phage_LUZ100_full_genome\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yostrich/anaconda3/bin/termseq_peaks\", line 63, in <module>\n",
      "    idr_peak_calls(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 527, in idr_peak_calls\n",
      "    oracle = mg.call_peaks().peaks_to_bed().moveto(oracle_fn)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 133, in peaks_to_bed\n",
      "    x = merge_narrowbed(x, self.strand)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 589, in merge_narrowbed\n",
      "    x = (peaks.cut([0, 1, 2, 4]).merge(c=4, **additional_kwargs)).to_dataframe()\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 923, in decorated\n",
      "    result = method(self, *args, **kwargs)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 402, in wrapped\n",
      "    stream = call_bedtools(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/helpers.py\", line 460, in call_bedtools\n",
      "    raise BEDToolsError(subprocess.list2cmdline(cmds), stderr)\n",
      "pybedtools.helpers.BEDToolsError: \n",
      "Command was:\n",
      "\n",
      "\tbedtools merge -o max -i /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.amb6ppbu.tmp -d 1 -c 4\n",
      "\n",
      "Error message was:\n",
      "\n",
      "*****\n",
      "***** ERROR: Requested column 4, but database file /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.amb6ppbu.tmp only has fields 1 - 0.\n",
      "\n",
      "Error: Unable to open file tssLUZ100/uninfectedEnriched-LUZ100_full-genome/uninfectedEnriched-LUZ100_full-genome.minus.peaks.oracle.narrowPeak. Exiting.\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.xrmr3yla.tmp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:tssLUZ100/uninfectedEnriched-NEB_control_spike_in/uninfectedEnriched-NEB_control_spike_in.plus.bedgraph\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:tssLUZ100/uninfectedEnriched-NEB_control_spike_in/uninfectedEnriched-NEB_control_spike_in.plus.bedgraph\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.ogtfu2k_.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.hlh3lvde.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.ogtfu2k_.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.hlh3lvde.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [1.64 0.20 0.94 0.45]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.xhb4n5y2.tmp\n",
      "INFO:root:in-vitro-RNA-transcript\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yostrich/anaconda3/bin/termseq_peaks\", line 63, in <module>\n",
      "    idr_peak_calls(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 527, in idr_peak_calls\n",
      "    oracle = mg.call_peaks().peaks_to_bed().moveto(oracle_fn)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 133, in peaks_to_bed\n",
      "    x = merge_narrowbed(x, self.strand)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/peaklib.py\", line 589, in merge_narrowbed\n",
      "    x = (peaks.cut([0, 1, 2, 4]).merge(c=4, **additional_kwargs)).to_dataframe()\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 923, in decorated\n",
      "    result = method(self, *args, **kwargs)\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/bedtool.py\", line 402, in wrapped\n",
      "    stream = call_bedtools(\n",
      "  File \"/home/yostrich/anaconda3/lib/python3.9/site-packages/pybedtools/helpers.py\", line 460, in call_bedtools\n",
      "    raise BEDToolsError(subprocess.list2cmdline(cmds), stderr)\n",
      "pybedtools.helpers.BEDToolsError: \n",
      "Command was:\n",
      "\n",
      "\tbedtools merge -o max -i /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.0b1s9teg.tmp -d 1 -c 4\n",
      "\n",
      "Error message was:\n",
      "\n",
      "*****\n",
      "***** ERROR: Requested column 4, but database file /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.0b1s9teg.tmp only has fields 1 - 0.\n",
      "\n",
      "Error: Unable to open file tssLUZ100/uninfectedEnriched-NEB_control_spike_in/uninfectedEnriched-NEB_control_spike_in.minus.peaks.oracle.narrowPeak. Exiting.\n",
      "mkdir: cannot create directory ‘tssLUZ100’: File exists\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.6kswrv23.tmp\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/uninfectedEnriched-PaLo41_Reference/uninfectedEnriched-PaLo41_Reference.plus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/uninfectedEnriched-PaLo41_Reference/uninfectedEnriched-PaLo41_Reference.plus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.v9fl7rmm.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.wm41i54j.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.v9fl7rmm.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.wm41i54j.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.08 0.20 0.94 0.41]\n",
      "INFO:root:Unioning bedgraphs...\n",
      "INFO:root:Averaging bedgraphs...\n",
      "INFO:root:/home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.1k1kz_5g.tmp\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/uninfectedEnriched-PaLo41_Reference/uninfectedEnriched-PaLo41_Reference.minus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:tssLUZ100/uninfectedEnriched-PaLo41_Reference/uninfectedEnriched-PaLo41_Reference.minus.bedgraph\n",
      "INFO:root:PaLo41_reference\n",
      "INFO:root:Merging peaks for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.c6ywx5h1.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.oj1ufw2u.tmp\n",
      "INFO:root:Calcluating IDR for /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.c6ywx5h1.tmp and /home/yostrich/anaconda3/lib/python3.9/site-packages/peaklib/pybedtools.oj1ufw2u.tmp\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.08 0.20 0.94 0.41]\n"
     ]
    }
   ],
   "source": [
    "# peak calling\n",
    "determinePeaks $BAMFiles tssLUZ100 \"5\" 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1645a29f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: dplyr\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# clustering using R\n",
    "chmod +x peak_clustering.r\n",
    "./peak_clustering.r \"tssLUZ100\" 20"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c528c257",
   "metadata": {},
   "source": [
    "# TTS detection and clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6b4fd1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "determinePeaks $BAMFiles ttsLUZ100 \"3\"\n",
    "# clustering using R\n",
    "chmod +x peak_clustering.r\n",
    "./peak_clustering.r \"ttsLUZ100\" 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "357b5ff3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
