{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d387eac",
   "metadata": {},
   "outputs": [],
   "source": [
    "# #!/bin/bash\n",
    "# #Input flags for batch files, do not run on notebook\n",
    "# while getopts 'i:f:' flag; do\n",
    "#   case \"${flag}\" in\n",
    "#       i) inputDirectory=\"${OPTARG}\";;\n",
    "#       f) inputDirectoryFastaFiles=${OPTARG} ;;\n",
    "#       ?) echo \"I don't know what flag is this\" ;;\n",
    "#            esac\n",
    "#        done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a96529a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#temporary assignment of variables for notebook, remove or comment out in bash file\n",
    "inputDirectory=LUZ7\n",
    "inputDirectoryFastaFiles=fastaFiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "789197fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#initialization of folder names\n",
    "output_folder_pychopper=output_pychopper\n",
    "output_alignment=BAM_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "df24e413",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘output_pychopper’: File exists\n",
      "mkdir: cannot create directory ‘output_pychopper/t5_test2’: File exists\n",
      "Using kit: PCS109\n",
      "Configurations to consider: \"+:SSP,-VNP|-:VNP,-SSP\"\n",
      "Counting fastq records in input file: LUZ7/t5_test2.fastq\n",
      "Total fastq records in input file: 125000\n",
      "Tuning the cutoff parameter (q) on 9938 sampled reads (8.0%) passing quality filters (Q >= 7.0).\n",
      "Optimizing over 30 cutoff values.\n",
      "100%|███████████████████████████████████████████| 30/30 [03:48<00:00,  7.61s/it]\n",
      "Best cutoff (q) value is 0.1724 with 91% of the reads classified.\n",
      "Processing the whole dataset using a batch size of 15625:\n",
      "100%|█████████████████████████████████| 125000/125000 [01:30<00:00, 1384.50it/s]\n",
      "Finished processing file: LUZ7/t5_test2.fastq\n",
      "Input reads failing mean quality filter (Q < 7.0): 0 (0.00%)\n",
      "Output fragments failing length filter (length < 50): 0\n",
      "-----------------------------------\n",
      "Reads with two primers:\t89.87%\n",
      "Rescued reads:\t\t0.30%\n",
      "Unusable reads:\t\t9.83%\n",
      "-----------------------------------\n"
     ]
    }
   ],
   "source": [
    "#run pychopper on fastq files in the inputdirectory.\n",
    "mkdir \"$output_folder_pychopper\"\n",
    "for file in \"$inputDirectory\"/*.fastq\n",
    "do\n",
    "    filename_extended=${file##*/}\n",
    "    filename=${filename_extended%.*}\n",
    "    mkdir \"$output_folder_pychopper\"/$filename\n",
    "    cdna_classifier.py \\\n",
    "        -r \"$output_folder_pychopper\"/\"$filename\"/${filename}_report.pdf \\\n",
    "        -u \"$output_folder_pychopper\"/\"$filename\"/${filename}_unclassified.fq \\\n",
    "        -S \"$output_folder_pychopper\"/\"$filename\"/${filename}_cdna_classifier_report.tsv \\\n",
    "        \"$file\" \\\n",
    "        \"$output_folder_pychopper\"/\"$filename\"/${filename}_full_length_output.fq\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7425759d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘temp’: File exists\n",
      "mkdir: cannot create directory ‘temp/cutadapt’: File exists\n",
      "This is cutadapt 3.7 with Python 3.9.7\n",
      "Command line parameters: -a A{10} -e 1 -j 0 -o temp/cutadapt/t5_test2_full_length_output_cutadapt.fq output_pychopper/t5_test2/t5_test2_full_length_output.fq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Done           00:00:03       112,502 reads @  27.1 µs/read;   2.22 M reads/minute\n",
      "Finished in 3.12 s (28 µs/read; 2.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 112,502\n",
      "Reads with adapters:                    63,877 (56.8%)\n",
      "Reads written (passing filters):       112,502 (100.0%)\n",
      "\n",
      "Total basepairs processed:    27,493,466 bp\n",
      "Total written (filtered):     25,643,127 bp (93.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AAAAAAAAAA; Type: regular 3'; Length: 10; Trimmed: 63877 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 32.6%\n",
      "  G: 23.6%\n",
      "  T: 43.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9557\t1757.8\t0\t9557\n",
      "4\t3395\t439.5\t0\t3395\n",
      "5\t2588\t109.9\t0\t2588\n",
      "6\t1845\t27.5\t0\t1845\n",
      "7\t1122\t6.9\t0\t1122\n",
      "8\t811\t1.7\t0\t811\n",
      "9\t1151\t0.4\t0\t1151\n",
      "10\t1885\t0.1\t1\t1071 814\n",
      "11\t1603\t0.1\t1\t954 649\n",
      "12\t1605\t0.1\t1\t998 607\n",
      "13\t1474\t0.1\t1\t1000 474\n",
      "14\t1436\t0.1\t1\t1043 393\n",
      "15\t1433\t0.1\t1\t1067 366\n",
      "16\t1402\t0.1\t1\t1140 262\n",
      "17\t1427\t0.1\t1\t1163 264\n",
      "18\t1400\t0.1\t1\t1184 216\n",
      "19\t1307\t0.1\t1\t1141 166\n",
      "20\t1473\t0.1\t1\t1337 136\n",
      "21\t1411\t0.1\t1\t1293 118\n",
      "22\t1379\t0.1\t1\t1303 76\n",
      "23\t1231\t0.1\t1\t1169 62\n",
      "24\t1263\t0.1\t1\t1222 41\n",
      "25\t1150\t0.1\t1\t1107 43\n",
      "26\t1076\t0.1\t1\t1032 44\n",
      "27\t1076\t0.1\t1\t1042 34\n",
      "28\t984\t0.1\t1\t954 30\n",
      "29\t909\t0.1\t1\t891 18\n",
      "30\t790\t0.1\t1\t770 20\n",
      "31\t807\t0.1\t1\t792 15\n",
      "32\t749\t0.1\t1\t739 10\n",
      "33\t671\t0.1\t1\t661 10\n",
      "34\t692\t0.1\t1\t680 12\n",
      "35\t629\t0.1\t1\t620 9\n",
      "36\t566\t0.1\t1\t561 5\n",
      "37\t556\t0.1\t1\t550 6\n",
      "38\t513\t0.1\t1\t504 9\n",
      "39\t486\t0.1\t1\t477 9\n",
      "40\t423\t0.1\t1\t417 6\n",
      "41\t412\t0.1\t1\t411 1\n",
      "42\t408\t0.1\t1\t403 5\n",
      "43\t361\t0.1\t1\t359 2\n",
      "44\t346\t0.1\t1\t343 3\n",
      "45\t333\t0.1\t1\t332 1\n",
      "46\t360\t0.1\t1\t352 8\n",
      "47\t296\t0.1\t1\t294 2\n",
      "48\t286\t0.1\t1\t282 4\n",
      "49\t228\t0.1\t1\t224 4\n",
      "50\t266\t0.1\t1\t262 4\n",
      "51\t234\t0.1\t1\t231 3\n",
      "52\t254\t0.1\t1\t250 4\n",
      "53\t205\t0.1\t1\t205\n",
      "54\t181\t0.1\t1\t177 4\n",
      "55\t185\t0.1\t1\t181 4\n",
      "56\t194\t0.1\t1\t189 5\n",
      "57\t155\t0.1\t1\t154 1\n",
      "58\t159\t0.1\t1\t156 3\n",
      "59\t155\t0.1\t1\t154 1\n",
      "60\t123\t0.1\t1\t123\n",
      "61\t161\t0.1\t1\t159 2\n",
      "62\t118\t0.1\t1\t118\n",
      "63\t131\t0.1\t1\t130 1\n",
      "64\t124\t0.1\t1\t123 1\n",
      "65\t85\t0.1\t1\t83 2\n",
      "66\t86\t0.1\t1\t86\n",
      "67\t85\t0.1\t1\t83 2\n",
      "68\t84\t0.1\t1\t84\n",
      "69\t81\t0.1\t1\t81\n",
      "70\t85\t0.1\t1\t84 1\n",
      "71\t74\t0.1\t1\t72 2\n",
      "72\t70\t0.1\t1\t68 2\n",
      "73\t61\t0.1\t1\t60 1\n",
      "74\t61\t0.1\t1\t58 3\n",
      "75\t56\t0.1\t1\t50 6\n",
      "76\t64\t0.1\t1\t58 6\n",
      "77\t51\t0.1\t1\t46 5\n",
      "78\t44\t0.1\t1\t42 2\n",
      "79\t66\t0.1\t1\t60 6\n",
      "80\t49\t0.1\t1\t45 4\n",
      "81\t53\t0.1\t1\t47 6\n",
      "82\t30\t0.1\t1\t28 2\n",
      "83\t34\t0.1\t1\t32 2\n",
      "84\t42\t0.1\t1\t37 5\n",
      "85\t34\t0.1\t1\t31 3\n",
      "86\t32\t0.1\t1\t31 1\n",
      "87\t40\t0.1\t1\t37 3\n",
      "88\t24\t0.1\t1\t22 2\n",
      "89\t33\t0.1\t1\t31 2\n",
      "90\t22\t0.1\t1\t22\n",
      "91\t32\t0.1\t1\t31 1\n",
      "92\t27\t0.1\t1\t25 2\n",
      "93\t26\t0.1\t1\t25 1\n",
      "94\t14\t0.1\t1\t14\n",
      "95\t19\t0.1\t1\t19\n",
      "96\t33\t0.1\t1\t33\n",
      "97\t14\t0.1\t1\t13 1\n",
      "98\t16\t0.1\t1\t16\n",
      "99\t15\t0.1\t1\t14 1\n",
      "100\t15\t0.1\t1\t15\n",
      "101\t13\t0.1\t1\t12 1\n",
      "102\t11\t0.1\t1\t11\n",
      "103\t16\t0.1\t1\t15 1\n",
      "104\t17\t0.1\t1\t15 2\n",
      "105\t10\t0.1\t1\t9 1\n",
      "106\t15\t0.1\t1\t12 3\n",
      "107\t15\t0.1\t1\t15\n",
      "108\t10\t0.1\t1\t10\n",
      "109\t14\t0.1\t1\t12 2\n",
      "110\t6\t0.1\t1\t3 3\n",
      "111\t12\t0.1\t1\t11 1\n",
      "112\t10\t0.1\t1\t10\n",
      "113\t7\t0.1\t1\t7\n",
      "114\t7\t0.1\t1\t6 1\n",
      "115\t9\t0.1\t1\t7 2\n",
      "116\t8\t0.1\t1\t8\n",
      "117\t7\t0.1\t1\t5 2\n",
      "118\t3\t0.1\t1\t3\n",
      "119\t8\t0.1\t1\t5 3\n",
      "120\t9\t0.1\t1\t7 2\n",
      "121\t8\t0.1\t1\t6 2\n",
      "122\t6\t0.1\t1\t4 2\n",
      "123\t6\t0.1\t1\t2 4\n",
      "124\t8\t0.1\t1\t6 2\n",
      "125\t4\t0.1\t1\t2 2\n",
      "126\t4\t0.1\t1\t3 1\n",
      "127\t8\t0.1\t1\t5 3\n",
      "128\t5\t0.1\t1\t3 2\n",
      "129\t7\t0.1\t1\t6 1\n",
      "130\t10\t0.1\t1\t5 5\n",
      "131\t5\t0.1\t1\t4 1\n",
      "132\t5\t0.1\t1\t3 2\n",
      "133\t3\t0.1\t1\t2 1\n",
      "134\t5\t0.1\t1\t4 1\n",
      "135\t7\t0.1\t1\t5 2\n",
      "136\t10\t0.1\t1\t6 4\n",
      "137\t3\t0.1\t1\t1 2\n",
      "138\t4\t0.1\t1\t4\n",
      "139\t5\t0.1\t1\t1 4\n",
      "140\t2\t0.1\t1\t1 1\n",
      "141\t3\t0.1\t1\t3\n",
      "142\t3\t0.1\t1\t2 1\n",
      "143\t3\t0.1\t1\t2 1\n",
      "144\t1\t0.1\t1\t0 1\n",
      "145\t2\t0.1\t1\t2\n",
      "146\t2\t0.1\t1\t0 2\n",
      "147\t4\t0.1\t1\t3 1\n",
      "148\t4\t0.1\t1\t3 1\n",
      "149\t1\t0.1\t1\t1\n",
      "150\t5\t0.1\t1\t3 2\n",
      "151\t3\t0.1\t1\t3\n",
      "152\t4\t0.1\t1\t3 1\n",
      "153\t6\t0.1\t1\t2 4\n",
      "154\t3\t0.1\t1\t0 3\n",
      "155\t5\t0.1\t1\t2 3\n",
      "156\t3\t0.1\t1\t0 3\n",
      "157\t3\t0.1\t1\t3\n",
      "158\t2\t0.1\t1\t0 2\n",
      "159\t1\t0.1\t1\t0 1\n",
      "160\t1\t0.1\t1\t1\n",
      "161\t1\t0.1\t1\t1\n",
      "162\t3\t0.1\t1\t2 1\n",
      "163\t2\t0.1\t1\t2\n",
      "164\t2\t0.1\t1\t1 1\n",
      "165\t2\t0.1\t1\t1 1\n",
      "166\t2\t0.1\t1\t1 1\n",
      "167\t2\t0.1\t1\t1 1\n",
      "168\t2\t0.1\t1\t1 1\n",
      "169\t4\t0.1\t1\t3 1\n",
      "170\t4\t0.1\t1\t0 4\n",
      "171\t2\t0.1\t1\t2\n",
      "172\t1\t0.1\t1\t1\n",
      "173\t2\t0.1\t1\t1 1\n",
      "174\t1\t0.1\t1\t0 1\n",
      "175\t1\t0.1\t1\t0 1\n",
      "177\t3\t0.1\t1\t2 1\n",
      "178\t1\t0.1\t1\t0 1\n",
      "179\t2\t0.1\t1\t1 1\n",
      "181\t2\t0.1\t1\t1 1\n",
      "184\t1\t0.1\t1\t1\n",
      "185\t5\t0.1\t1\t3 2\n",
      "186\t3\t0.1\t1\t3\n",
      "188\t3\t0.1\t1\t2 1\n",
      "190\t3\t0.1\t1\t1 2\n",
      "191\t1\t0.1\t1\t1\n",
      "193\t2\t0.1\t1\t0 2\n",
      "194\t3\t0.1\t1\t1 2\n",
      "195\t2\t0.1\t1\t0 2\n",
      "197\t2\t0.1\t1\t0 2\n",
      "198\t2\t0.1\t1\t0 2\n",
      "199\t3\t0.1\t1\t1 2\n",
      "200\t2\t0.1\t1\t1 1\n",
      "201\t2\t0.1\t1\t1 1\n",
      "202\t2\t0.1\t1\t2\n",
      "204\t2\t0.1\t1\t1 1\n",
      "206\t1\t0.1\t1\t1\n",
      "207\t1\t0.1\t1\t1\n",
      "209\t1\t0.1\t1\t0 1\n",
      "214\t2\t0.1\t1\t0 2\n",
      "215\t1\t0.1\t1\t1\n",
      "217\t1\t0.1\t1\t0 1\n",
      "218\t1\t0.1\t1\t0 1\n",
      "221\t1\t0.1\t1\t1\n",
      "223\t1\t0.1\t1\t1\n",
      "225\t2\t0.1\t1\t1 1\n",
      "226\t1\t0.1\t1\t1\n",
      "227\t1\t0.1\t1\t0 1\n",
      "228\t5\t0.1\t1\t2 3\n",
      "229\t1\t0.1\t1\t0 1\n",
      "230\t1\t0.1\t1\t0 1\n",
      "233\t1\t0.1\t1\t0 1\n",
      "234\t2\t0.1\t1\t0 2\n",
      "235\t5\t0.1\t1\t2 3\n",
      "237\t1\t0.1\t1\t0 1\n",
      "238\t1\t0.1\t1\t0 1\n",
      "239\t2\t0.1\t1\t0 2\n",
      "240\t1\t0.1\t1\t0 1\n",
      "241\t2\t0.1\t1\t0 2\n",
      "242\t1\t0.1\t1\t0 1\n",
      "243\t2\t0.1\t1\t0 2\n",
      "244\t1\t0.1\t1\t0 1\n",
      "245\t4\t0.1\t1\t1 3\n",
      "246\t6\t0.1\t1\t1 5\n",
      "247\t6\t0.1\t1\t0 6\n",
      "248\t5\t0.1\t1\t0 5\n",
      "249\t6\t0.1\t1\t0 6\n",
      "250\t9\t0.1\t1\t1 8\n",
      "251\t10\t0.1\t1\t0 10\n",
      "252\t12\t0.1\t1\t0 12\n",
      "253\t9\t0.1\t1\t0 9\n",
      "254\t11\t0.1\t1\t2 9\n",
      "255\t11\t0.1\t1\t1 10\n",
      "256\t10\t0.1\t1\t1 9\n",
      "257\t14\t0.1\t1\t2 12\n",
      "258\t18\t0.1\t1\t1 17\n",
      "259\t6\t0.1\t1\t0 6\n",
      "260\t11\t0.1\t1\t0 11\n",
      "261\t7\t0.1\t1\t0 7\n",
      "262\t11\t0.1\t1\t0 11\n",
      "263\t11\t0.1\t1\t0 11\n",
      "264\t6\t0.1\t1\t0 6\n",
      "265\t14\t0.1\t1\t1 13\n",
      "266\t3\t0.1\t1\t0 3\n",
      "267\t3\t0.1\t1\t0 3\n",
      "268\t2\t0.1\t1\t0 2\n",
      "269\t1\t0.1\t1\t0 1\n",
      "270\t4\t0.1\t1\t1 3\n",
      "271\t3\t0.1\t1\t0 3\n",
      "272\t3\t0.1\t1\t0 3\n",
      "273\t1\t0.1\t1\t0 1\n",
      "274\t3\t0.1\t1\t0 3\n",
      "275\t1\t0.1\t1\t0 1\n",
      "276\t2\t0.1\t1\t0 2\n",
      "277\t3\t0.1\t1\t0 3\n",
      "278\t3\t0.1\t1\t0 3\n",
      "280\t2\t0.1\t1\t0 2\n",
      "281\t4\t0.1\t1\t0 4\n",
      "282\t1\t0.1\t1\t0 1\n",
      "283\t4\t0.1\t1\t0 4\n",
      "284\t2\t0.1\t1\t0 2\n",
      "285\t5\t0.1\t1\t0 5\n",
      "286\t4\t0.1\t1\t0 4\n",
      "287\t7\t0.1\t1\t0 7\n",
      "288\t6\t0.1\t1\t0 6\n",
      "289\t6\t0.1\t1\t0 6\n",
      "290\t8\t0.1\t1\t0 8\n",
      "291\t9\t0.1\t1\t0 9\n",
      "292\t14\t0.1\t1\t0 14\n",
      "293\t18\t0.1\t1\t0 18\n",
      "294\t17\t0.1\t1\t0 17\n",
      "295\t27\t0.1\t1\t0 27\n",
      "296\t27\t0.1\t1\t0 27\n",
      "297\t37\t0.1\t1\t0 37\n",
      "298\t31\t0.1\t1\t0 31\n",
      "299\t37\t0.1\t1\t0 37\n",
      "300\t49\t0.1\t1\t0 49\n",
      "301\t53\t0.1\t1\t0 53\n",
      "302\t70\t0.1\t1\t0 70\n",
      "303\t62\t0.1\t1\t0 62\n",
      "304\t71\t0.1\t1\t0 71\n",
      "305\t53\t0.1\t1\t0 53\n",
      "306\t88\t0.1\t1\t0 88\n",
      "307\t74\t0.1\t1\t0 74\n",
      "308\t87\t0.1\t1\t0 87\n",
      "309\t76\t0.1\t1\t0 76\n",
      "310\t48\t0.1\t1\t0 48\n",
      "311\t74\t0.1\t1\t0 74\n",
      "312\t53\t0.1\t1\t0 53\n",
      "313\t47\t0.1\t1\t0 47\n",
      "314\t44\t0.1\t1\t1 43\n",
      "315\t44\t0.1\t1\t0 44\n",
      "316\t34\t0.1\t1\t0 34\n",
      "317\t43\t0.1\t1\t0 43\n",
      "318\t20\t0.1\t1\t0 20\n",
      "319\t19\t0.1\t1\t0 19\n",
      "320\t26\t0.1\t1\t0 26\n",
      "321\t24\t0.1\t1\t0 24\n",
      "322\t6\t0.1\t1\t0 6\n",
      "323\t7\t0.1\t1\t0 7\n",
      "324\t12\t0.1\t1\t0 12\n",
      "325\t5\t0.1\t1\t0 5\n",
      "326\t4\t0.1\t1\t0 4\n",
      "327\t3\t0.1\t1\t0 3\n",
      "328\t5\t0.1\t1\t0 5\n",
      "329\t4\t0.1\t1\t0 4\n",
      "330\t3\t0.1\t1\t0 3\n",
      "331\t2\t0.1\t1\t0 2\n",
      "333\t3\t0.1\t1\t0 3\n",
      "334\t1\t0.1\t1\t0 1\n",
      "335\t2\t0.1\t1\t0 2\n",
      "336\t1\t0.1\t1\t0 1\n",
      "338\t2\t0.1\t1\t0 2\n",
      "339\t4\t0.1\t1\t0 4\n",
      "340\t1\t0.1\t1\t0 1\n",
      "341\t1\t0.1\t1\t0 1\n",
      "342\t1\t0.1\t1\t0 1\n",
      "343\t1\t0.1\t1\t0 1\n",
      "344\t3\t0.1\t1\t0 3\n",
      "345\t2\t0.1\t1\t1 1\n",
      "348\t2\t0.1\t1\t0 2\n",
      "350\t1\t0.1\t1\t0 1\n",
      "351\t1\t0.1\t1\t0 1\n",
      "352\t1\t0.1\t1\t0 1\n",
      "354\t2\t0.1\t1\t0 2\n",
      "355\t1\t0.1\t1\t0 1\n",
      "362\t1\t0.1\t1\t1\n",
      "363\t1\t0.1\t1\t0 1\n",
      "364\t1\t0.1\t1\t0 1\n",
      "365\t1\t0.1\t1\t0 1\n",
      "367\t1\t0.1\t1\t1\n",
      "371\t2\t0.1\t1\t0 2\n",
      "394\t1\t0.1\t1\t0 1\n",
      "400\t1\t0.1\t1\t0 1\n",
      "401\t1\t0.1\t1\t0 1\n",
      "406\t1\t0.1\t1\t0 1\n",
      "411\t1\t0.1\t1\t0 1\n",
      "415\t1\t0.1\t1\t0 1\n",
      "422\t1\t0.1\t1\t0 1\n",
      "428\t1\t0.1\t1\t0 1\n",
      "439\t1\t0.1\t1\t0 1\n",
      "443\t1\t0.1\t1\t0 1\n",
      "448\t2\t0.1\t1\t0 2\n",
      "450\t1\t0.1\t1\t0 1\n",
      "451\t1\t0.1\t1\t0 1\n",
      "458\t1\t0.1\t1\t0 1\n",
      "483\t1\t0.1\t1\t0 1\n",
      "500\t1\t0.1\t1\t1\n",
      "518\t1\t0.1\t1\t0 1\n",
      "519\t1\t0.1\t1\t0 1\n",
      "524\t1\t0.1\t1\t0 1\n",
      "525\t1\t0.1\t1\t0 1\n",
      "570\t1\t0.1\t1\t0 1\n",
      "584\t1\t0.1\t1\t0 1\n",
      "585\t1\t0.1\t1\t0 1\n",
      "612\t1\t0.1\t1\t1\n",
      "662\t1\t0.1\t1\t0 1\n",
      "677\t1\t0.1\t1\t0 1\n",
      "709\t1\t0.1\t1\t0 1\n",
      "1090\t1\t0.1\t1\t1\n",
      "1121\t1\t0.1\t1\t0 1\n",
      "This is cutadapt 3.7 with Python 3.9.7\n",
      "Command line parameters: -g TTTCTGTTGGTGCTGATATTGCTGGG -e 1 -j 0 -o temp/cutadapt/t5_test2_full_length_output_cutadapt_2.fq temp/cutadapt/t5_test2_full_length_output_cutadapt.fq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Done           00:00:03       112,502 reads @  27.1 µs/read;   2.21 M reads/minute\n",
      "Finished in 3.06 s (27 µs/read; 2.21 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 112,502\n",
      "Reads with adapters:                    37,603 (33.4%)\n",
      "Reads written (passing filters):       112,502 (100.0%)\n",
      "\n",
      "Total basepairs processed:    25,643,127 bp\n",
      "Total written (filtered):     25,528,780 bp (99.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TTTCTGTTGGTGCTGATATTGCTGGG; Type: regular 5'; Length: 26; Trimmed: 37603 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-25 bp: 0; 26 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t36582\t1757.8\t0\t36582\n",
      "4\t660\t439.5\t0\t660\n",
      "5\t234\t109.9\t0\t234\n",
      "6\t116\t27.5\t0\t116\n",
      "7\t7\t6.9\t0\t7\n",
      "8\t1\t1.7\t0\t1\n",
      "10\t2\t0.1\t0\t2\n",
      "18\t1\t0.0\t0\t1\n"
     ]
    }
   ],
   "source": [
    "#runs cutadapt to remove the poly-A tails and SSP sequences\n",
    "mkdir temp\n",
    "mkdir temp/cutadapt\n",
    "for file in \"$output_folder_pychopper\"/*/*_full_length_output.fq\n",
    "do\n",
    "    filename_extended=${file##*/}\n",
    "    searchstring=\"_full_length_output.fq\"\n",
    "    filename=${filename_extended%$searchstring*}\n",
    "    ~/.local/bin/cutadapt -a A{10} -e 1 -j 0 -o temp/cutadapt/${filename}_full_length_output_cutadapt.fq ${file}\n",
    "    ~/.local/bin/cutadapt -g TTTCTGTTGGTGCTGATATTGCTGGG -e 1 -j 0 -o temp/cutadapt/${filename}_full_length_output_cutadapt_2.fq temp/cutadapt/${filename}_full_length_output_cutadapt.fq\n",
    "    rm temp/cutadapt/${filename}_full_length_output_cutadapt.fq\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e5643810",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘temp/minimap2’: File exists\n",
      "mkdir: cannot create directory ‘temp/minimap2/LUZ7’: File exists\n",
      "[M::mm_idx_gen::0.018*2.58] collected minimizers\n",
      "[M::mm_idx_gen::0.039*2.80] sorted minimizers\n",
      "[M::main::0.039*2.78] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.041*2.64] mid_occ = 3\n",
      "[M::mm_idx_stat] kmer size: 14; skip: 10; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.043*2.52] distinct minimizers: 13730 (99.75% are singletons); average occurrences: 1.003; average spacing: 5.401\n",
      "[M::worker_pipeline::1.616*3.46] mapped 112502 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -k14 -t 8 fastaFiles/LUZ7.fasta temp/cutadapt/t5_test2_full_length_output_cutadapt_2.fq\n",
      "[M::main] Real time: 1.636 sec; CPU: 5.609 sec; Peak RSS: 0.081 GB\n",
      "mkdir: cannot create directory ‘temp/minimap2/US449’: File exists\n",
      "[M::mm_idx_gen::0.299*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.386*1.58] sorted minimizers\n",
      "[M::main::0.386*1.58] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.408*1.57] mid_occ = 16\n",
      "[M::mm_idx_stat] kmer size: 14; skip: 10; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.423*1.52] distinct minimizers: 957321 (85.04% are singletons); average occurrences: 1.245; average spacing: 5.420\n",
      "[M::worker_pipeline::3.027*3.88] mapped 112502 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -k14 -t 8 fastaFiles/US449.fasta temp/cutadapt/t5_test2_full_length_output_cutadapt_2.fq\n",
      "[M::main] Real time: 3.040 sec; CPU: 11.766 sec; Peak RSS: 0.162 GB\n",
      "mkdir: cannot create directory ‘temp/minimap2/in-vitro-RNA-transcript’: File exists\n",
      "[M::mm_idx_gen::0.002*9.09] collected minimizers\n",
      "[M::mm_idx_gen::0.003*4.67] sorted minimizers\n",
      "[M::main::0.003*4.47] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.004*4.37] mid_occ = 2\n",
      "[M::mm_idx_stat] kmer size: 14; skip: 10; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.004*4.30] distinct minimizers: 373 (100.00% are singletons); average occurrences: 1.000; average spacing: 5.316\n",
      "[M::worker_pipeline::1.338*1.58] mapped 112502 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -ax map-ont -k14 -t 8 fastaFiles/in-vitro-RNA-transcript.fasta temp/cutadapt/t5_test2_full_length_output_cutadapt_2.fq\n",
      "[M::main] Real time: 1.366 sec; CPU: 2.141 sec; Peak RSS: 0.069 GB\n"
     ]
    }
   ],
   "source": [
    "#maps the preprocessed reads onto the given genome(s)\n",
    "mkdir temp/minimap2\n",
    "for file in temp/cutadapt/*.fq\n",
    "do\n",
    "    for fastaFile in $inputDirectoryFastaFiles/*.fasta\n",
    "    do\n",
    "        filename_extended=${file##*/}\n",
    "        searchstring=\"_full_length_output_cutadapt_2.fq\"\n",
    "        filename=${filename_extended%$searchstring*}\n",
    "        fastaname_extended=${fastaFile##*/}\n",
    "        fastaname=${fastaname_extended%.*}\n",
    "        mkdir temp/minimap2/${fastaname}\n",
    "        minimap2 -ax map-ont -k14 -t 8 ${fastaFile} ${file} > temp/minimap2/${fastaname}/\"${filename}-${fastaname}_alignment.sam\"\n",
    "    done\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "158819c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#creates an index for the given genomic fasta files\n",
    "for fastaFile in $inputDirectoryFastaFiles/*.fasta\n",
    "do\n",
    "    samtools faidx ${fastaFile}\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5836c6d0",
   "metadata": {},
   "source": [
    "# SAM to BAM conversion + indexing with clipping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "962f297a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d9bfad10",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘BAM_files’: File exists\n",
      "mkdir: cannot create directory ‘temp/SAM_and_BAM’: File exists\n",
      "[samclip] samclip 0.4.0 by Torsten Seemann (@torstenseemann)\n",
      "[samclip] Loading: fastaFiles/LUZ7.fasta.fai\n",
      "[samclip] Found 1 sequences in fastaFiles/LUZ7.fasta.fai\n",
      "[samclip] Processed 100000 records...\n",
      "[samclip] Total SAM records 112514, removed 4546, allowed 16114, passed 107968\n",
      "[samclip] Header contained 2 lines\n",
      "[samclip] Done.\n",
      "[samclip] samclip 0.4.0 by Torsten Seemann (@torstenseemann)\n",
      "[samclip] Loading: fastaFiles/LUZ7.fasta.fai\n",
      "[samclip] Found 1 sequences in fastaFiles/LUZ7.fasta.fai\n",
      "[samclip] Processed 100000 records...\n",
      "[samclip] Total SAM records 112513, removed 4546, allowed 16116, passed 107967\n",
      "[samclip] Header contained 2 lines\n",
      "[samclip] Done.\n",
      "[samclip] samclip 0.4.0 by Torsten Seemann (@torstenseemann)\n",
      "[samclip] Loading: fastaFiles/US449.fasta.fai\n",
      "[samclip] Found 1 sequences in fastaFiles/US449.fasta.fai\n",
      "[samclip] Processed 100000 records...\n",
      "[samclip] Total SAM records 135322, removed 12391, allowed 51189, passed 122931\n",
      "[samclip] Header contained 2 lines\n",
      "[samclip] Done.\n",
      "[samclip] samclip 0.4.0 by Torsten Seemann (@torstenseemann)\n",
      "[samclip] Loading: fastaFiles/US449.fasta.fai\n",
      "[samclip] Found 1 sequences in fastaFiles/US449.fasta.fai\n",
      "[samclip] Processed 100000 records...\n",
      "[samclip] Total SAM records 135371, removed 12434, allowed 51283, passed 122937\n",
      "[samclip] Header contained 2 lines\n",
      "[samclip] Done.\n",
      "[samclip] samclip 0.4.0 by Torsten Seemann (@torstenseemann)\n",
      "[samclip] Loading: fastaFiles/in-vitro-RNA-transcript.fasta.fai\n",
      "[samclip] Found 1 sequences in fastaFiles/in-vitro-RNA-transcript.fasta.fai\n",
      "[samclip] Processed 100000 records...\n",
      "[samclip] Total SAM records 112502, removed 23, allowed 65, passed 112479\n",
      "[samclip] Header contained 2 lines\n",
      "[samclip] Done.\n",
      "[samclip] samclip 0.4.0 by Torsten Seemann (@torstenseemann)\n",
      "[samclip] Loading: fastaFiles/in-vitro-RNA-transcript.fasta.fai\n",
      "[samclip] Found 1 sequences in fastaFiles/in-vitro-RNA-transcript.fasta.fai\n",
      "[samclip] Processed 100000 records...\n",
      "[samclip] Total SAM records 112502, removed 23, allowed 65, passed 112479\n",
      "[samclip] Header contained 2 lines\n",
      "[samclip] Done.\n"
     ]
    }
   ],
   "source": [
    "#removes clips (10bp) using samclip, creates a BAM file from the SAM files whilst also removing the reads with a MAXQ score\n",
    "#lower than 30.\n",
    "mkdir ${output_alignment}\n",
    "mkdir temp/SAM_and_BAM\n",
    "for file in temp/minimap2/*/*.sam\n",
    "do\n",
    "    filename_extended=${file##*/}\n",
    "    searchstring=\"_alignment.sam\"\n",
    "    filename=${filename_extended%$searchstring*}\n",
    "    tempString=$(echo \"$file\" |awk -F'temp/minimap2/|/*.sam' '{print $2}')\n",
    "    fastaName=$(echo $tempString | cut -f 1 -d\"/\")\n",
    "    \n",
    "    samclip --max 10 --ref  $inputDirectoryFastaFiles/${fastaName}.fasta < ${file} > temp/SAM_and_BAM/${filename}_clipped.sam\n",
    "    samtools view -q 30 -S -b temp/SAM_and_BAM/${filename}_clipped.sam > temp/SAM_and_BAM/${filename}_clipped.bam \n",
    "    samtools sort temp/SAM_and_BAM/${filename}_clipped.bam -o  ${output_alignment}/${filename}.sorted.bam\n",
    "    samtools index ${output_alignment}/${filename}.sorted.bam\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ca7c1731",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[E::sam_parse1] SEQ and QUAL are of different length\n",
      "[W::sam_read1] Parse error at line 3650441\n",
      "[main_samview] truncated file.\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff59ec2f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
